### PR TITLE
Fix parser of optional fields in SAM/BAM.

### PR DESCRIPTION
**Pull request by alumi**
Fixed returned type of following two elements in optional fields in SAM/BAM.

|type|description|module|type (before)|type (after)|
|:-:|:-:|:-:|:-:|:-:|
|A|Printable character|bam|byte|char|
|H|Byte array in the Hex format|sam|nil|byte-array|

### CODE REVIEW

File: src/cljam/io/sam/util.clj [hunk 1]
```diff
@@ -77,6 +77,19 @@
         (map parse-header-line)
         into-header)))
 
+(defn- ^"[B" as-byte-array [^String s]
```
Comment (totakke) [inline src/cljam/io/sam/util.clj]: proton has `hex->bytes` function, which seems to be equivalent of this function. Is any reason to reinvent it?

Comment (alumi) [inline src/cljam/io/sam/util.clj]: I hadn't noticed that fn. Thank you for pointing that!

**Comment by totakke** [comment]
Thank you :)

**Comment by alumi** [comment]
Thank you for the reviewing!

**Pull request merged**